### ISSUE TITLE
Do not require a return value from handlers throughout the SDK (if possible)

### ISSUE THREAD

**Issue by eladb** [winglang/wing#4188]
### I tried this:

```js
bring cloud;
bring http;

let api = new cloud.Api();

api.get("/", inflight () => {
  // no return statement
});

test "main" {
  let res = http.get(api.url);
  log("${res.status}");
}
```

### This happened:

```
500
```

### I expected this:

I think in this case we should simply return an empty body and a `200 OK` status.

We should change the handler's signature to return an *optional* and default to `{ status: 200 }`.

We should also do that for any other closure signature where it makes sense.

### Component

SDK

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by Chriscbr** [comment]
I think the higher-order bit here is that the compiler should be raising a type checking error since the inflight function passed to `api.get` expects to return an `ApiResponse` object -- this is a more broad error that affects user code beyond SDK usage. Here's a simpler example:

```js
let f: inflight (): num = inflight () => {
  log("not returning anything");
};

test "call f" {
  let x = f();
  log("${x + 3}"); // prints NaN today
}
```

----

@eladb Are there other HTTP frameworks that return 200 responses by default you were inspired by / set precedent for this kind of behavior?

(I'm also wondering whether the default behaviors for handling requests/responses be something users may want to configure with middleware?)

**Comment by garysassano** [comment]
I agree with this change. When creating a new Lambda function, by default it has the following handler:

```
export const handler = async (event) => {
  // TODO implement
  const response = {
    statusCode: 200,
    body: JSON.stringify('Hello from Lambda!'),
  };
  return response;
};
```
So it makes sense that we reproduce a similar behaviour when creating an empty `cloud.Function`.

**Comment by eladb** [comment]
I think returning 200 for success and 500 for a thrown exception is a very reasonable default behavior for an http handler.

I believe node's http server returns 200 by default but I am not sure.

**Comment by eladb** [comment]
Please bump to p1

**Comment by staycoolcall911** [comment]
Done

**Comment by staycoolcall911** [comment]
@eladb did it for the most popular resources
